lib/rucio/core/rse.py

```diff
@@ -740,14 +747,12 @@ def get_rse_limits(rse_id, name=None, session=None):
     query = session.query(models.RSELimit).filter_by(rse_id=rse_id)
     if name:
         query = query.filter_by(name=name)
-    limits = {}
-    for limit in query:
-        limits[limit.name] = limit.value
-    return limits
+    return {limit.name: limit.value for limit in query}
 
 
 @transactional_session
-def delete_rse_limits(rse_id, name=None, session=None):
+def delete_rse_limits(rse_id: str, name: 'Optional[str]' = None,
+                      session: 'Optional[Session]' = None) -> None:
```

Comment:
technically the session type is not `Optional`, since the function decoration always fills it when it is `None`. For static type checking reasons within the function, I would remove the `Optional` (see also above).